openmeter/streaming/clickhouse/group_by_values_query.go

```diff
@@ -0,0 +1,52 @@
+package clickhouse
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/huandu/go-sqlbuilder"
+
+	meterpkg "github.com/openmeterio/openmeter/openmeter/meter"
+)
+
+type listGroupByValuesQuery struct {
+	Database        string
+	EventsTableName string
+	Namespace       string
+	Meter           meterpkg.Meter
+	GroupByKey      string
+	From            *time.Time
+	To              *time.Time
+	Search          *string
+}
+
+func (d listGroupByValuesQuery) toSQL() (string, []interface{}) {
+	tableName := getTableName(d.Database, d.EventsTableName)
+
+	// Helper function to get the column
+	getColumn := columnFactory(d.EventsTableName)
+
+	groupByJSONPath := d.Meter.GroupBy[d.GroupByKey]
+
+	sb := sqlbuilder.ClickHouse.NewSelectBuilder()
+	sb.Select(sb.As(fmt.Sprintf("DISTINCT JSON_VALUE(%s, '%s')", getColumn("data"), escapeJSONPathLiteral(groupByJSONPath)), "group_by_values"))
```

Comment:
would `query.Var(groupByJSONPath)` work here to move to arg?

Reply:
In `Select`, yes. I'll move it.